### ISSUE TITLE
YouTube Back Broken

### ISSUE THREAD

**Issue by Baron-Severin** [mozilla-mobile/firefox-tv#1939]
## Steps to reproduce
- Launch app
- Open YouTube tile
- Press back repeatedly

### Expected behavior
- App should return to home screen after focus moves to YouTube side bar

### Actual behavior
- App never returns to home screen

### Device information
* Fire TV device: Cube, emulator
* Latest affected Firefox version: ac9c612b5e4c44af98a25d6906f962668b0f9e6c (head of master)
* Earliest affected Firefox version: c9f19ecb131be8f313b2c8efc918b794c106b8e1

### Debugging Notes
- Commenting out the following line in `MainActivity#onCreate` fixes the problem:
```kotlin
webRenderComponents.sessionManager.getOrCreateEngineSession().resetView(this@MainActivity)
```
- The proximate cause of this bug is that, when we attempt to back from YouTube, "app:home" is not in the backstack (we expect that it would be)
  - Note that this means that the following steps still work as expected
```
* Launch app
* Open IMDB tile
* Open Menu
* Open YouTube Tile
* Press back repeatedly
(success)
```

**Comment by Baron-Severin** [comment]
Replacing `webRenderComponents.sessionManager.getOrCreateEngineSession().resetView(this@MainActivity)`
with 
`webRenderComponents.sessionManager.getOrCreateEngineSession()`
 fixes the bug.  So this is specifically related to resetting the view, and not just a change in initialization order.

**Comment by liuche** [comment]
When this gets fixed, please cherry-pick this onto the 3.5-LAT2 branch.

**Comment by levisacal** [comment]
wasn't able to reproduce anymore in 3.5-LAt on 4k pendant
